src/Arcus.WebApi.Security.Authorization/Jwt/JwtTokenReader.cs

```diff
@@ -14,18 +14,50 @@ public class JwtTokenReader : IJwtTokenReader
             "https://login.microsoftonline.com/common/v2.0/.well-known/openid-configuration";
 
         private readonly string _applicationId;
+        private readonly TokenValidationParameters _tokenValidationParameters;
         private readonly JwtSecurityTokenHandler _handler;
         private readonly ConfigurationManager<OpenIdConnectConfiguration> _configManager;
 
-        public JwtTokenReader(string applicationId)
+        /// <summary>
+        ///     Constructor
+        /// </summary>
+        /// <remarks>Uses Microsoft OpenId connect discovery endpoint</remarks>
+        /// <param name="applicationId">Azure AD Application used as audience to validate against</param>
+        public JwtTokenReader(string applicationId) : this()
         {
             Guard.NotNullOrWhitespace(applicationId, nameof(applicationId));
 
             _applicationId = applicationId;
+            _configManager = new ConfigurationManager<OpenIdConnectConfiguration>(MicrosoftDiscoveryEndpoint, new OpenIdConnectConfigurationRetriever());
+        }
+
+        /// <summary>
+        ///     Constructor
+        /// </summary>
+        /// <remarks>Uses Microsoft OpenId connect discovery endpoint</remarks>
+        /// <param name="tokenValidationParameters">Collection of parameters to influence how the token validation is done</param>
+        public JwtTokenReader(TokenValidationParameters tokenValidationParameters) : this(tokenValidationParameters, MicrosoftDiscoveryEndpoint)
+        {
+        }
 
+        /// <summary>
+        ///     Constructor
+        /// </summary>
+        /// <param name="openIdConnectDiscoveryUri">Uri of an OpenId connect endpoint for discovering the configuration</param>
+        /// <param name="tokenValidationParameters">Collection of parameters to influence how the token validation is done</param>
+        public JwtTokenReader(TokenValidationParameters tokenValidationParameters, string openIdConnectDiscoveryUri) :
+            this()
+        {
+            Guard.NotNullOrWhitespace(openIdConnectDiscoveryUri, nameof(openIdConnectDiscoveryUri));
+            Guard.NotNull(tokenValidationParameters, nameof(tokenValidationParameters));
+
+            _tokenValidationParameters = tokenValidationParameters;
+            _configManager = new ConfigurationManager<OpenIdConnectConfiguration>(openIdConnectDiscoveryUri, new OpenIdConnectConfigurationRetriever());
+        }
+
+        private JwtTokenReader()
```

Comment:
I find it a bit tricky to guide every ctor overload to this overload, just to set the field. It's easily forgotton I guess.
If it's needed anyway, why not make it static?

Reply:
Good idea

Reply:
I'm in favor of initializing it outside of the ctor.  However, don't know if it can be static.   It depends a bit on what this JwtSecurityHandler instance does.   Does it keep state ?  In that case, it is tricky to declare it as static imo.

Reply:
I'll initialize it on the field without static!